### ISSUE TITLE
🐛 Bug Report: azure integration is not using visibility secret annotation on secrets

### ISSUE THREAD

**Issue by ppiwow-apay** [backstage/backstage#20174]
### 📜 Description

Hi, it seems that azure integration config is not using https://backstage.io/docs/conf/defining/#visibility and because of that secrets are shown as a plain text in https://github.com/backstage/backstage/blob/master/plugins/devtools/README.md#config   
Problaby somewhere here? https://github.com/backstage/backstage/blob/5b1daa2c41e51c581f3f8df0c161701f878fa3c3/packages/integration/src/azure/config.ts#L95

### 👍 Expected behavior

```json
integrations ":{1 item
"azure": [1 item
    0: {
        2 items "host": string "dev.azure.com"
        "credential": {
            3 items "clientId": string "plain-text"
            "clientSecret":   string"<secret>"
            "tenantId": string "plain-text"
        }
    }
]
}
```

### 👎 Actual Behavior with Screenshots

```json
integrations ":{1 item
"azure": [1 item
    0: {
        2 items "host": string "dev.azure.com"
        "credential": {
            3 items "clientId": string "plain-text"
            "clientSecret": string "plain-text"
            "tenantId": string "plain-text"
        }
    }
]
}
```

### 👟 Reproduction steps

Just install plugin 

https://github.com/backstage/backstage/blob/master/plugins/devtools/README.md#config

put the section about integration in app-config.yaml

go to config tab in dev tools

### 🖥️ Your Environment

OS:   Windows_NT 10.0.22621 - win32/x64
node: v18.17.1
yarn: 1.22.19
cli:  0.22.13 (installed)
backstage:  1.18.1

Dependencies:
  @backstage/app-defaults                                          1.4.3
  @backstage/backend-app-api                                       0.5.3
  @backstage/backend-common                                        0.15.2, 0.19.5
  @backstage/backend-dev-utils                                     0.1.1
  @backstage/backend-openapi-utils                                 0.0.4
  @backstage/backend-plugin-api                                    0.6.3
  @backstage/backend-tasks                                         0.5.8
  @backstage/catalog-client                                        1.4.4
  @backstage/catalog-model                                         1.4.2
  @backstage/cli-common                                            0.1.12
  @backstage/cli-node                                              0.1.4
  @backstage/cli                                                   0.22.13
  @backstage/config-loader                                         1.5.0
  @backstage/config                                                1.1.0
  @backstage/core-app-api                                          1.10.0
  @backstage/core-components                                       0.11.2, 0.13.5
  @backstage/core-plugin-api                                       1.6.0
  @backstage/errors                                                1.2.2
  @backstage/eslint-plugin                                         0.1.3
  @backstage/integration-aws-node                                  0.1.6
  @backstage/integration-react                                     1.1.19
  @backstage/integration                                           1.7.0
  @backstage/plugin-api-docs                                       0.9.11
  @backstage/plugin-app-backend                                    0.3.51
  @backstage/plugin-app-node                                       0.1.3
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.1.0
  @backstage/plugin-auth-backend-module-github-provider            0.1.0
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.0
  @backstage/plugin-auth-backend-module-google-provider            0.1.0
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.0
  @backstage/plugin-auth-backend                                   0.19.0
  @backstage/plugin-auth-node                                      0.2.19, 0.3.0
  @backstage/plugin-azure-devops-backend                           0.4.0
  @backstage/plugin-azure-devops-common                            0.3.1
  @backstage/plugin-azure-devops                                   0.3.6
  @backstage/plugin-catalog-backend-module-azure                   0.1.22
  @backstage/plugin-catalog-backend-module-msgraph                 0.5.10
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.0
  @backstage/plugin-catalog-backend-module-unprocessed             0.3.0
  @backstage/plugin-catalog-backend                                1.13.0
  @backstage/plugin-catalog-common                                 1.0.16
  @backstage/plugin-catalog-graph                                  0.2.36
  @backstage/plugin-catalog-import                                 0.10.0
  @backstage/plugin-catalog-node                                   1.4.4
  @backstage/plugin-catalog-react                                  1.8.4
  @backstage/plugin-catalog-unprocessed-entities                   0.1.3
  @backstage/plugin-catalog                                        1.13.0
  @backstage/plugin-devtools-backend                               0.2.0
  @backstage/plugin-devtools-common                                0.1.4
  @backstage/plugin-devtools                                       0.1.4
  @backstage/plugin-events-node                                    0.2.12
  @backstage/plugin-github-actions                                 0.6.5
  @backstage/plugin-home-react                                     0.1.3
  @backstage/plugin-home                                           0.5.8
  @backstage/plugin-kubernetes-backend                             0.12.0
  @backstage/plugin-kubernetes-common                              0.6.6
  @backstage/plugin-kubernetes                                     0.10.3
  @backstage/plugin-org                                            0.6.14
  @backstage/plugin-pagerduty                                      0.6.5
  @backstage/plugin-permission-backend                             0.5.26
  @backstage/plugin-permission-common                              0.7.8
  @backstage/plugin-permission-node                                0.7.14
  @backstage/plugin-permission-react                               0.4.15
  @backstage/plugin-proxy-backend                                  0.4.0
  @backstage/plugin-scaffolder-backend                             1.17.0
  @backstage/plugin-scaffolder-common                              1.4.1
  @backstage/plugin-scaffolder-node                                0.2.3
  @backstage/plugin-scaffolder-react                               1.5.5
  @backstage/plugin-scaffolder                                     1.15.0
  @backstage/plugin-search-backend-module-catalog                  0.1.7
  @backstage/plugin-search-backend-module-pg                       0.5.12
  @backstage/plugin-search-backend-module-techdocs                 0.1.7
  @backstage/plugin-search-backend-node                            1.2.7
  @backstage/plugin-search-backend                                 1.4.3
  @backstage/plugin-search-common                                  1.2.6
  @backstage/plugin-search-react                                   1.7.0
  @backstage/plugin-search                                         1.4.0
  @backstage/plugin-tech-insights-backend-module-jsonfc            0.1.35
  @backstage/plugin-tech-insights-backend                          0.5.17
  @backstage/plugin-tech-insights-common                           0.2.12
  @backstage/plugin-tech-insights-node                             0.4.9
  @backstage/plugin-tech-insights                                  0.3.16
  @backstage/plugin-tech-radar                                     0.6.8
  @backstage/plugin-techdocs-backend                               1.7.0
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.0
  @backstage/plugin-techdocs-node                                  1.8.0
  @backstage/plugin-techdocs-react                                 1.1.11
  @backstage/plugin-techdocs                                       1.7.0
  @backstage/plugin-user-settings                                  0.7.10
  @backstage/release-manifests                                     0.0.10
  @backstage/test-utils                                            1.4.3
  @backstage/theme                                                 0.2.19, 0.4.2
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.5

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by Rugvip** [comment]
I'm unable to reproduce this, it seems to behave correctly with this minimal config:

```yaml
app:
  baseUrl: 'x'
backend:
  listen: 5000
  baseUrl: 'x'
  database:
    client: better-sqlite3
    connection: ':memory:'
techdocs:
  builder: local

integrations:
  azure:
    - host: test
      credential:
        clientSecret: test
```

```log
$ yarn backstage-cli config:print --lax --config mini-config.yaml --package @backstage/backend-common

Loaded config from mini-config.yaml
app:
  baseUrl: x
backend:
  listen: 5000
  baseUrl: x
  database:
    client: better-sqlite3
    connection: <secret>
techdocs:
  builder: local
integrations:
  azure:
    - host: test
      credential:
        clientSecret: <secret>
```

**Comment by detox3r** [comment]
idk if related, but I jus t created a brand new backstage app and just replaced the azure integration, (tried it with service principal and pat token in ADO):

```yaml
app:
  title: Scaffolded Backstage App
  baseUrl: http://localhost:3000

organization:
  name: My Company

backend:
  baseUrl: http://localhost:7007
  listen:
    port: 7007
  csp:
    connect-src: ["'self'", 'http:', 'https:']
  cors:
    origin: http://localhost:3000
    methods: [GET, HEAD, PATCH, POST, PUT, DELETE]
    credentials: true
  database:
    client: better-sqlite3
    connection: ':memory:'

integrations:
  azure:
    - host: dev.azure.com
      credentials:
        - clientId: xxxxxxxxxxxxxxxxxx
          clientSecret: xxxxxxxxxxxxxx
          tenantId: xxxxxxxxxxxxxxxxx

```
so the error says:

```
index.esm.js:232 Failed to read configuration array at 'credentials' as it does not have any visible elements. See https://backstage.io/docs/conf/defining#visibility for instructions on how to make it visible.
```

**Comment by djflocko** [comment]
I have the same problem: when printing the configuration, the personalAccessToken is not marked as secret. It is visible and plain text. In the DevTools plugin it is also displayed as a plain string. Any idea ?

**Comment by fstoerkle** [comment]
This will actually be fixed with the 1.19.0 release, see https://github.com/backstage/backstage/pull/20039.
The above reproduction case behaves expected when upgrading Backstage to `1.19.0-next.1`

**Comment by Rugvip** [comment]
@fstoerkle great, thank you! Will close this
